flocker/node/agents/functional/test_gce.py

```diff
@@ -78,7 +81,18 @@ def create_instance(self, instance_name):
             instance_name,
             machine_type=u"f1-micro"
         )
-        self.addCleanup(lambda: instance.destroy())
+
+        def destroy_best_effort(inst):
+            try:
+                inst.destroy()
+            except HttpError as e:
+                if e.resp.status == 404:
+                    # The test must have already destroyed the instance.
+                    pass
+                else:
+                    raise e
```

Comment:
Thanks you for the reminder! When I wrote that I felt like I was doing it wrong.